docs/Reference/CLI/CLI-Subcommands.md

```diff
@@ -59,3 +59,314 @@ Generate a list of peer IDs including the private key, public key, and peer ID w
 * `--outputFile` is the path and filename of the file to which the keys are written. If not specified, the peer IDs
 are written to the `./config/peer-ids.dat` file. 
 * `<number` is the number of peer IDs to generate.
+
+## validator
+
+Register or generate validators and send transactions to an Ethereum 1.0 node.
+
+### generate
+
+Register a validator by generating new keys and sending deposit transactions to an Ethereum 1.0
+node.
+
+#### deposit-amount-gwei
+
+```bash tab="Syntax"
+teku validator generate --deposit-amount-gwei=<GWEI>
+```
+
+```bash tab="Example"
+teku validator generate --deposit-amount-gwei=32000000000
+```
+
+Amount to deposit in the Ethereum 1.0 deposit contract.
+
+#### encrypted-keystore-enabled
+
+```bash tab="Syntax"
+teku validator generate --encrypted-keystore-enabled=<BOOLEAN>
+```
+
+```bash tab="Example"
+teku validator generate --encrypted-keystore-enabled=false
+```
+
+Specify whether to create encrypted V3 keystore validator and withdrawal keys.
```

Comment:
We do not create "encrypted v3 keystore". We create "encrypted BLS12-381 keystore" as per [EIP-2335](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2335.md) spec.

Reply:
Technically it's v4 I think.